### PR TITLE
Fix: validate input or throw when using MessagesPlaceholder

### LINKED ISSUE #2871: Better error message for invalid input on `MessagesPlaceholder`
When trying to pass content to `MessagesPlaceholder` and it's not of type `Message` we should throw a better error

### PR DESCRIPTION
**Pull request by bracesproul**
Closes #2871 

Throw when input passed to `MessagesPlaceholder.formatMessages` is not of correct type

### CODE REVIEW

File: langchain/src/prompts/chat.ts [hunk 1]
```diff
@@ -158,9 +158,33 @@ export class MessagesPlaceholder<
...
 
+  validateInputOrThrow(input: Array<unknown>): input is BaseMessage[] {
+    let isInputBaseMessage = false;
+
+    if (Array.isArray(input)) {
+      isInputBaseMessage = input.every((message) =>
+        isBaseMessage(message as BaseMessage)
+      );
+    } else {
+      isInputBaseMessage = isBaseMessage(input as BaseMessage);
```
Comment (jacoblee93) [inline langchain/src/prompts/chat.ts:169]: Can you change `isBaseMessage` to tolerate `null/undefined` values?

https://github.com/langchain-ai/langchainjs/blob/main/langchain/src/schema/index.ts#L461

File: langchain/src/prompts/chat.ts [hunk 2]
```diff
@@ -158,9 +158,33 @@ export class MessagesPlaceholder<
...
+    }
+
+    if (!isInputBaseMessage) {
+      const readableInput =
+        typeof input === "string" ? input : JSON.stringify(input, null, 2);
+      throw new Error(
```
Comment (jacoblee93) [inline langchain/src/prompts/chat.ts:175]: I would also add something to the error message about which input the array of messages is expected for

Comment (bracesproul) [reply langchain/src/prompts/chat.ts:185]: Refactored to `Class 'MessagesPlaceholder' expects an array of BaseMessage instances but received:...`

Comment (jacoblee93) [reply langchain/src/prompts/chat.ts:185]: And actually maybe add that specifically `MessagesPlaceholder expects...`

Comment (bracesproul) [reply langchain/src/prompts/chat.ts:175]: Updated

Comment (jacoblee93) [reply langchain/src/prompts/chat.ts:175]: Can we do: `Error: Field "${variableName}" in prompt uses a MessagesPlaceholder, which expects an array of BaseMessages as an input value. Received: ${readableInput}`

Comment (jacoblee93) [reply langchain/src/prompts/chat.ts:175]: And then also pass `variableName` into the validator?

Comment (bracesproul) [reply langchain/src/prompts/chat.ts:175]: Updated
```typescript
throw new Error(
        `Error: Field "${variableName}" in prompt uses a MessagesPlaceholder, which expects an array of BaseMessages as an input value. Received: ${readableInput}`
      );
```
and `validateInputOrThrow` now takes in two inputs, first the unknown value, and second the variable name

**Comment by jacoblee93** [comment]
Thank you!

**Review by jacoblee93 — APPROVED** [review]

**Pull request merged**